### PR TITLE
Section Fronts: Give Sandbox

### PR DESCRIPTION
**Pull request by nypl-wluisi**
**Sandbox (Backend):** 
https://pr999-nypl1.pantheonsite.io/user/login?bypass

**Sandbox (Frontend):** 
https://scout-git-section-fronts-give-nypl.vercel.app/give

**This PR does the following:**
- [RENO-3182](https://jira.nypl.org/browse/RENO-3182): Adds `Donation` component for Give page 
- [RENO-3208](https://jira.nypl.org/browse/RENO-3208): Adds `CatchAllRoutesPage` component to pages.
- [RENO-3205](https://jira.nypl.org/browse/RENO-3205): Adds NextJS Preview api route and functionality
- [RENO-3219](https://jira.nypl.org/browse/RENO-3219): Add On Demand Static Regeneration

### Review Steps:
- [ ] Pull in branch
- [ ] Set `DRUPAL_API` env variable to point to Multidev: `https://pr999-nypl1.pantheonsite.io`
- [ ] Start a local build in production mode: `npm run build && npm start`

### CODE REVIEW

File: src/components/shared/CardGrid/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  layout,
+  isCentered,
+  isBordered = false,
+}: CardProps) {
+  let imagePropsFinal = {};
+
+  if (layout === "row") {
+    imagePropsFinal = {
+      // Workaround to suppress console.warn message for component and aspect ratio.
+      aspectRatio: "square",
```
Comment (isastettler) [inline src/components/shared/CardGrid/Card.tsx:48]: Interesting. Why is it "square" and not "TwoByOne"?

File: src/components/shared/CardGrid/CardGridHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+export default function CardGridHeader({
+  id,
+  title,
+  headingColor = "brand.primary",
+  link,
+  hrefText,
+}: CardGridHeaderProps) {
+  if (link && hrefText && title) {
```
Comment (isastettler) [inline src/components/shared/CardGrid/CardGridHeader.tsx:25]: Is this use case on page? Or is this in preparation of other pages?

File: src/components/section-fronts/SectionFrontPage/SectionFrontPage.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  }
+`;
+
+interface SectionFrontPageProps {
+  uuid: string;
+  isPreview?: boolean;
+  revisionId?: string;
+}
+
+// @TODO I dont think you need isPreview and revisionId inside this component, and it never gets called?
```
[the remark this answers is not in the document]
Comment (nypl-wluisi) [reply src/components/section-fronts/SectionFrontPage/SectionFrontPage.tsx:105]: It was more of a note to myself, but I think I can remove, as it's actually incorrect. I thought initially it was redundant to have those values in there, but they're actually needed. We can discuss more.

File: src/components/shared/CardGrid/CardGridHeader.tsx (hunk 2 above)
Comment (isastettler) [reply src/components/shared/CardGrid/CardGridHeader.tsx:25]: Oh I meant the link on the title.

**Pull request merged**